Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableAssetRef.cpp

```diff
@@ -0,0 +1,117 @@
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &SpawnableAssetRef::m_asset, "m_asset", "")
+                    ->Attribute(AZ::Edit::Attributes::ShowProductAssetFileName, false)
+                    ->Attribute(AZ::Edit::Attributes::HideProductFilesInAssetPicker, true)
+                    ->Attribute(AZ::Edit::Attributes::AssetPickerTitle, "Spawnable Asset")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &SpawnableAssetRef::OnSpawnAssetChanged);
+            }
+        }
+
+        if (auto behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+        {
+            behaviorContext
+                ->Class<SpawnableAssetRef>("SpawnableAssetRef")
+                ->Constructor()
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+                ->Attribute(AZ::Script::Attributes::Category, "Prefab/Spawning")
+                ->Attribute(AZ::Script::Attributes::Module, "prefabs")
+                ->Property("m_asset", BehaviorValueProperty(&SpawnableAssetRef::m_asset));
+        }
+    }
+
+    SpawnableAssetRef::SpawnableAssetRef()
+    {
+    }
+
+    SpawnableAssetRef::~SpawnableAssetRef()
+    {
+    }
+
+    SpawnableAssetRef::SpawnableAssetRef(const SpawnableAssetRef& rhs)
+        : m_asset(rhs.m_asset)
+    {
+    }
+
+    SpawnableAssetRef& SpawnableAssetRef::operator=(const SpawnableAssetRef& rhs)
+    {
+        m_asset = rhs.m_asset;
+        return *this;
+    }
+
+    void SpawnableAssetRef::OnSpawnAssetChanged()
+    {
+        if (m_asset.GetId().IsValid())
+        {
+            AZStd::string rootSpawnableFile;
+            StringFunc::Path::GetFileName(m_asset.GetHint().c_str(), rootSpawnableFile);
+
+            rootSpawnableFile += Spawnable::DotFileExtension;
+
+            AZ::u32 rootSubId = SpawnableAssetHandler::BuildSubId(move(rootSpawnableFile));
+
+            if (m_asset.GetId().m_subId != rootSubId)
+            {
+                AZ::Data::AssetId rootAssetId = m_asset.GetId();
+                rootAssetId.m_subId = rootSubId;
+
+                m_asset = AZ::Data::AssetManager::Instance().FindOrCreateAsset<Spawnable>(
+                    rootAssetId, AZ::Data::AssetLoadBehavior::Default);
+            }
+            else
+            {
+                m_asset.SetAutoLoadBehavior(AZ::Data::AssetLoadBehavior::Default);
+            }
+
+            AZ::Data::AssetBus::Handler::BusDisconnect();
+            AZ::Data::AssetBus::Handler::BusConnect(m_asset.GetId());
+        }
+    }
+
+    void SpawnableAssetRef::OnAssetReady([[maybe_unused]] AZ::Data::Asset<AZ::Data::AssetData> asset)
+    {
+    }
+
+    void SpawnableAssetRef::OnAssetReloaded(AZ::Data::Asset<AZ::Data::AssetData> asset)
+    {
+        m_asset = asset;
+    }
+
+    void SpawnableAssetRef::OnAssetUnloaded(
+        [[maybe_unused]] const AZ::Data::AssetId assetId,
+        [[maybe_unused]] const AZ::Data::AssetType assetType)
+    {
+    }
```

Comment:
```suggestion
```

Reply:
fixed.. this was accidently leftover from debugging